aiohttp/_cookie_helpers.py

```diff
@@ -185,7 +188,33 @@ def parse_cookie_header(header: str) -> list[tuple[str, Morsel[str]]]:
         # Use the same pattern as parse_set_cookie_headers to find cookies
         match = _COOKIE_PATTERN.match(header, i)
         if not match:
-            break
+            # Fallback for malformed cookies (issue #11632)
+            # Find next semicolon to skip or attempt simple key=value parsing
+            next_semi = header.find(";", i)
+            eq_pos = header.find("=", i)
+
+            # Try to extract key=value if '=' comes before ';'
+            if eq_pos != -1 and (next_semi == -1 or eq_pos < next_semi):
+                end_pos = next_semi if next_semi != -1 else n
+                key = header[i:eq_pos].strip()
+                value = header[eq_pos + 1 : end_pos].strip()
+
+                # Validate the name (same as regex path)
+                if not key or not _COOKIE_NAME_RE.match(key):
+                    if key:  # Only warn if there was a key to validate
+                        internal_logger.warning(
+                            "Can not load cookie: Illegal cookie name %r", key
+                        )
+                else:
```

Comment:
Maybe simpler like this?
```suggestion
                if not _COOKIE_NAME_RE.match(key):
                    internal_logger.warning(
                        "Can not load cookie: Illegal cookie name %r", key
                    )
                elif key:
```

Reply:
2ed4cfc9082f74f3c919f865029498d439bb51dd